azurerm/resource_arm_role_assignment.go

```diff
@@ -56,6 +58,10 @@ func resourceArmRoleAssignmentCreate(d *schema.ResourceData, meta interface{}) e
 	roleDefinitionId := d.Get("role_definition_id").(string)
 	principalId := d.Get("principal_id").(string)
 
+	if name == "" {
+		name = uuid.NewV1().String()
```

Comment:
Minor-ish nit here, but we should be using v4 UUIDs to ensure they are as random as possible.